openmc/deplete/nuclide.py

```diff
@@ -211,14 +213,190 @@ def to_xml_element(self):
+        Array ``(n_energy, n_products)`` where
+        ``yield_matrix[g, j]`` is the fission yield of product
+        ``j`` for energy group ``g``.
+
+    See Also
+    --------
+    :meth:`from_xml_element`, :meth:`from_dict`
+    """
+
+    def __init__(self, ordered_energies, ordered_products, group_fission_yields):
+        check_type("energies", ordered_energies, Iterable, Real)
+        self.energies = tuple(ordered_energies)
+        check_type("products", ordered_products, Iterable, str)
+        self.products = tuple(ordered_products)
+        yield_matrix = asarray(group_fission_yields, dtype=float)
+        if yield_matrix.shape != (len(self.energies), len(self.products)):
+            raise ValueError(
+                "Shape of yield matrix inconsistent. "
+                "Should be ({}, {}), is {}".format(
+                    len(ordered_energies), len(ordered_products),
+                    yield_matrix.shape))
+        self.yield_matrix = yield_matrix
+
+    def __len__(self):
+        return len(self.energies)
+
+    def __getitem__(self, energy):
+        if energy not in self.energies:
+            raise KeyError(energy)
+        return _FissionYield(
+            self.products, self.yield_matrix[self.energies.index(energy)])
+
+    def __iter__(self):
+        return iter(self.energies)
+
+    @classmethod
+    def from_xml_element(cls, element):
+        """Construct a distribution from a depletion chain xml file
+
+        Parameters
+        ----------
+        element : xml.etree.ElementTree.Element
+            XML element to pull fission yield data from
+
+        Returns
+        -------
+        FissionYieldDistribution
+        """
+        yields = {}
+        for elem_index, yield_elem in enumerate(element.iter("fission_yields")):
+            energy = float(yield_elem.get("energy"))
+            products = yield_elem.find("products").text.split()
+            yield_mapobj = map(float, yield_elem.find("data").text.split())
+            # Get a map of products to their corresponding yield
+            yields[energy] = dict(zip(products, yield_mapobj))
+
+        return cls.from_dict(yields)
+
+    @classmethod
+    def from_dict(cls, fission_yields):
+        """Construct a distribution from a dictionary of yields
+
+        Parameters
+        -----------
+        fission_yields : dict
+            Dictionary ``{energy: {product: yield}}``
+
+        Returns
+        -------
+        FissionYieldDistribution
+        """
+        # mapping {energy: {product: value}}
+        energies = tuple(sorted(fission_yields))
+
+        # Get a consistent set of products to produce a matrix of yields
+        shared_prod = set()
+        for prod_set in map(set, fission_yields.values()):
+            shared_prod |= prod_set
+        ordered_prod = tuple(sorted(shared_prod))
+
+        yield_matrix = empty((len(energies), len(shared_prod)))
+
+        for g_index, energy in enumerate(sorted(energies)):
```

Comment:
energies is already sorted
```suggestion
        for g_index, energy in enumerate(energies):
```